configserver/src/test/java/com/yahoo/vespa/config/server/ConfigServerBootstrapTest.java

```diff
@@ -81,6 +88,33 @@ public void testBootstrap() throws Exception {
         assertFalse(vipStatus.isInRotation());
     }
 
+    // Just tests setup, the actual response of accessing /status.html depends on the status
+    // file existing or not, which cannot be tested here
+    @Test
+    public void testBootstrapWithVipStatusFile() throws Exception {
+        ConfigserverConfig configserverConfig = createConfigserverConfig(temporaryFolder);
+        InMemoryProvisioner provisioner = new InMemoryProvisioner(true, "host0", "host1", "host3");
+        DeployTester tester = new DeployTester(configserverConfig, provisioner);
+        tester.deployApp("src/test/apps/hosted/");
+
+        File versionFile = temporaryFolder.newFile();
+        VersionState versionState = new VersionState(versionFile);
+        assertTrue(versionState.isUpgraded());
+
+        RpcServer rpcServer = createRpcServer(configserverConfig);
+        VipStatus vipStatus = createVipStatus(VIP_STATUS_FILE);
+        ConfigServerBootstrap bootstrap = new ConfigServerBootstrap(tester.applicationRepository(), rpcServer,
+                                                                    versionState, createStateMonitor(),
+                                                                    vipStatus, INITIALIZE_ONLY, VIP_STATUS_FILE);
+        assertTrue(vipStatus.isInRotation()); // default is in rotation when using status file
```

Comment:
This is for backwards compatibility? Won't this make it impossible for us to use VIP_STATUS_FILE safely?

Reply:
This is what VipStatus thinks about the status, if a ststus file is used the response is based on whether the file exists or not (see comment for the test).

Reply:
Got it, 👍